app/controllers/tags_controller.rb

```diff
@@ -123,29 +123,21 @@ def csv_upload
     # Gets parameters for the upload
     file = params[:csv_tags]
     encoding = params[:encoding]
-
-    if request.post? && !file.blank?
-      begin
-        Tag.transaction do
-          invalid_lines = []
-          nb_updates = Tag.parse_csv(file,
-                                     @current_user,
-                                     invalid_lines,
-                                     encoding)
-          unless invalid_lines.empty?
-            flash[:error] = I18n.t('csv_invalid_lines') +
-                            invalid_lines.join(', ')
-          end
-          if nb_updates > 0
-            flash[:success] = I18n.t('tags.upload.upload_success',
-                                     nb_updates: nb_updates)
-          end
+    if file
+      Tag.transaction do
+        result = MarkusCSV.parse(file.read, encoding: encoding) do |row|
+          next if CSV.generate_line(row).strip.empty?
```

Comment:
I don't like using `next` here, the block should really just deal line by line. Why not keep an `if` check and put the following line in the "then" part?